src/lib/critical_entry/Pin.svelte

```diff
@@ -1,39 +1,28 @@
 <script lang="ts">
-  import { Marker, type LngLat, type MapMouseEvent } from "maplibre-gl";
+  import { Marker, type LngLat } from "maplibre-gl";
   import { map } from "stores";
   import { onDestroy, onMount } from "svelte";
 
-  export let enableAdding: boolean;
-
   let marker: Marker | null = null;
-  export let markerPosition: LngLat | null = null;
+  export let markerPosition: LngLat;
+  export let markerPositionUpdated: () => any;
```

Comment:
Hmm, needing a callback feels like a code smell, since Svelte should do reactivity for us. I guess the problem is somewhere in binding a member of an array during `#each` and not having that update make it look like the entire array has changed? Kind of hard to know without trying something out

Reply:
I agree. The issue was that I didn't know how to make the lines redraw  on the map live without it.